From 2a3732f1575d69b64ed5c9b76c6213c7d4fbc20b Mon Sep 17 00:00:00 2001
From: Adam Sampson <ats@offog.org>
Date: Tue, 8 Sep 2020 15:55:12 +0100
Subject: [PATCH] Fix arr_strlcpy call in ifp_device.

directory_chooser needs to know the length of the target.
---
 src/ifp_device.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/ifp_device.c b/src/ifp_device.c
index 9ef5741..42236cb 100644
--- a/src/ifp_device.c
+++ b/src/ifp_device.c
@@ -824,7 +824,7 @@ gchar *npath;
 
 
 void
-directory_chooser(char * title, GtkWidget * parent, char * directory) {
+directory_chooser(char * title, GtkWidget * parent, char * directory, size_t directory_len) {
 
         GtkWidget * dialog;
 	const gchar * selected_directory;
@@ -855,7 +855,7 @@ directory_chooser(char * title, GtkWidget * parent, char * directory) {
 			gtk_widget_destroy(dialog);
 		}
 
-                arr_strlcpy(directory, selected_directory);
+                g_strlcpy(directory, selected_directory, directory_len);
 		g_free(utf8);
         }
 
@@ -865,7 +865,7 @@ directory_chooser(char * title, GtkWidget * parent, char * directory) {
 
 void
 local_path_selected_cb(GtkButton * button, gpointer data) {
-	directory_chooser(_("Please select a local path."), aifp_window, dest_dir);
+	directory_chooser(_("Please select a local path."), aifp_window, dest_dir, sizeof dest_dir);
         gtk_entry_set_text(GTK_ENTRY(local_path_entry), dest_dir);
 }
 
